packages/web-react/src/components/Button/Button.tsx

```diff
@@ -1,47 +1,78 @@
-import React from 'react';
+import React, { MouseEvent } from 'react';
 import classNames from 'classnames';
+import { WithChildren } from '../../types/main';
 
 type Color = 'primary' | 'secondary' | 'tertiary';
 
 const getButtonColorClassname = (color: Color): string => `Button--${color}`;
 
-export interface ButtonProps {
+export interface ButtonProps extends WithChildren {
   /**
    * Emotion color variant.
    */
   color: Color;
-  /**
-   * Button label.
-   */
-  label: string;
   /**
    * Function to be called on button click.
    */
-  onClick: () => void;
+  onClick?: (event: MouseEvent) => void;
   /**
    * HTML `type` attribute.
    */
   type: 'button' | 'submit';
+  /**
+   * Block alignment modification
+   */
+  block: boolean;
+  /**
+   * Disabling button
+   */
+  disabled: boolean;
+  /**
+   * Aria label
+   */
+  ariaLabel?: string;
 }
 
 export const Button = ({
   color,
-  label,
   onClick,
   type,
-}: ButtonProps): JSX.Element => (
-  <button
-    className={classNames('Button', getButtonColorClassname(color))}
-    onClick={onClick}
-    type={type}
-  >
-    {label}
-  </button>
-);
+  block,
+  ariaLabel,
+  disabled,
+  ...restAttributes
+}: ButtonProps): JSX.Element => {
+  const handleClick = (event: MouseEvent) => {
+    if (disabled) {
+      event.preventDefault();
+
+      return;
+    }
+
+    if (typeof onClick === 'function') {
+      onClick(event);
+    }
+  };
+
+  return (
+    <button
```

Comment:
Also you should use `children` prop here to be more transparent and not using React magic.

Reply:
its inspired by reactstrap implementation https://github.com/reactstrap/reactstrap/blob/master/src/Button.js I don't think it's necessary to pass on children

Reply:
Yes, it is not necessary to pass it, but it is more obvious and also better for readability of code to use it. Because you must be acquainted of React magic behind using children prop. And it is also confusing when you are defining a prop and not using it in component.

Reply:
Well, if you want to develop in react, you have to be familiar with how it works ... you can also recognize it by the interface `export interface ButtonProps extends WithChildren `

Reply:
at the same time, documentation, examples and snapshot tests should help in the future.